docs/source/endpoints/login.md

```diff
@@ -0,0 +1,71 @@
+---
+myst:
+  html_meta:
+    "description": "The @login endpoint exposes the list of external authentication services that may be used in the Plone site."
+    "property=og:description": "The @login endpoint exposes the list of external authentication services that may be used in the Plone site."
+    "property=og:title": "@login for external authentication links"
+    "keywords": "Plone, plone.restapi, REST, API, login, authentication, external services"
+---
+
+# Login for external authentication links
+
+It is common to use add-ons that allow logging in to your site using third party services.
+Such add-ons include using authentication services provided by KeyCloak, GitHub, or other OAuth2 or OpenID Connect enabled services.
+
+When you install one of these add-ons, it modifies the login process, directing the user to third party services.
+
+To expose the links provided by these add-ons, `plone.restapi` provides an adapter based service registration.
+It lets those add-ons know that the REST API can use those services to authenticate users.
+This will mostly be used by frontends that need to show the end user the links to those services.
+
+To achieve that, third party products need to register one or more adapters for the Plone site root object, providing the `plone.restapi.interfaces.IExternalLoginProviders` interface.
+
+In the adapter, the add-on needs to return the list of external links and some metadata, including the `id`, `title`, and name of the `plugin`.
```

Comment:
What's the difference between `id` and `plugin`? Why do we need both?

Reply:
This is on purpose to support the case of a plugin added twice. Think of 2 `oidc` plugins in a site. They would be different in `id` but same in `plugin`, like a way to mark the "plugin type" in `plugin`.

Reply:
That's what I do, for instance, in my PR on pas.plugins.oidc: https://github.com/collective/pas.plugins.oidc/blob/use-adapter/src/pas/plugins/oidc/services/login/get.py#L23-L26

Reply:
@erral Got it. I didn't understand from the docs what `plugin` meant. I'll update the example below to show "pas.plugins.authomatic" as the plugin to make it a bit clearer.